keyvault/client/src/main/java/com/azure/keyvault/SecretAsyncClient.java

```diff
@@ -0,0 +1,462 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault;
+
+import com.azure.common.ServiceClient;
+import com.azure.common.http.HttpPipeline;
+import com.azure.common.http.rest.PagedResponse;
+import com.azure.common.http.rest.Response;
+import com.azure.common.http.rest.SimpleResponse;
+import com.azure.common.http.rest.VoidResponse;
+import com.azure.common.implementation.RestProxy;
+import com.azure.keyvault.implementation.SecretAttributesPage;
+import com.azure.keyvault.models.DeletedSecret;
+import com.azure.keyvault.models.Secret;
+import com.azure.keyvault.models.SecretAttributes;
+import org.apache.commons.lang3.Validate;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URL;
+import java.util.Objects;
+import java.util.function.Function;
+
+
+/**
+ * The {@link SecretAsyncClient secretAsyncClient} provides asynchronous methods to manage to {@link Secret secrets} in the Azure Key Vault.
+ * This client uses reactor to communicate asynchonously with Azure Key Vault. It uses {@link Mono wrapper} to return a single {@link Secret secret} response
+ * and {@link Flux wrapper} to return multiple {@link Secret secrets} in cases of list operations.
+ *
+ * <p>A {@link Secret secret} is a resource managed by Key Vault. It is represented by non-null fields secret.name and secret.value.
+ * The secret.expires, secret.contentType and secret.notBefore values in the {@link Secret secret} are optional. The secret.enabled
+ * field is set to true by default in Azure Key Vault, if not specified. The secret.id, secret.created, secret.updated, secret.recoveryLevel
+ * fields are auto assigned when the secret is created in the key vault.</p>
+ *
+ * <p>Samples to construct {@link SecretAsyncClient}</p>
+ * <pre>
+ *    SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+ *                                .vaultEndpoint("https://myvault.vault.azure.net/")
+ *                                .credentials(keyVaultCredentials)
```

Comment:
Currently, at this point we don't supply KeyVaultCredentials as part of the API.
We expect the user to implement AsyncServiceClientCredentials or ServiceClientCredentials and create their credentials class. Then instantiate it and pass it in.

I feel, with Scott's MSAL, we will be providing a KeyVaultCredentials class as part of API, it would be worth adding that to Javadocs, when we have Scott's version in the API implemented.
Added a TODO in code for it.